hugolib/site.go

```diff
@@ -1355,7 +1356,7 @@ func (s *Site) buildSiteMeta() (err error) {
 
 	s.assembleMenus()
 
-	s.Info.LastChange = s.Pages[0].Lastmod
+	s.Info.LastChange = s.Pages.ByLastmod()[ln-1].Lastmod
```

Comment:
1) I suspect there may be other and conflicting date setting somewhere else
2) Sorting the pages to get the last date sounds ineffective
3) Test